houston/recorder.py

```diff
@@ -0,0 +1,75 @@
+__all__ = ['Recorder']
+
+from typing import Type, FrozenSet, Dict, Any, List
+import attr
+import logging
+
+import json
+import threading
+
+from .state import State
+
+logger = logging.getLogger(__name__)  # type: logging.Logger
+logger.setLevel(logging.DEBUG)
+
+
+class NoFileSetError(BaseException):
+    """
+    Thrown when writing to file is requested but filename is not
+    set.
+    """
+    def __init__(self):
+        super().__init__("No file is set for the recorder.")
+
+
+class Recorder(object):
+    """
+    Describes a Recorder object that keeps tracks of states
```

Comment:
No need for `Describes a Recorder object`. Rewrite to: `Recorders are used to ...`.
Also, some extra documentation explaining the methods for this class would be useful.